File: app/src/main/java/com/stevenfrew/beatprompter/filter/TagFilter.java
package com.stevenfrew.beatprompter.filter;

import com.stevenfrew.beatprompter.cache.SongFile;

import java.util.ArrayList;

public class TagFilter extends SongFilter
{
    public TagFilter(String tag,ArrayList<SongFile> songs)
    {
        super(tag,songs,true);
    }

    public boolean equals(Object o) {
        return o instanceof TagFilter && mName.equals(((TagFilter) o).mName);
    }
}

File: app/src/main/java/com/stevenfrew/beatprompter/filter/TagFilter.kt
package com.stevenfrew.beatprompter.filter

import com.stevenfrew.beatprompter.cache.SongFile
import java.util.ArrayList

class TagFilter(tag: String, songs: ArrayList<SongFile>) : SongFilter(tag, songs, true) {

    override fun equals(other: Any?): Boolean {
        return other is TagFilter && mName == other.mName
    }
}